### PR TITLE
ft/deeplinking-link-component

### PR DESCRIPTION
**Pull request by thompsongl**
Adds component to allow for wrapping `deepLinking` links

If accepted, would allow for using `wrapComponents` to alter/override links related to `deepLinking`. We use ui-router for application routing and the `deepLinking` hash-based scheme doesn't mesh well. Therefore, we'll bypass standard `deepLinking` routes and use ui-router instead, while taking advantage of swagger-ui plugins.

### CODE REVIEW

**Comment by heldersepu** [comment]
I find the name `DeepLinkingLink` a bit redundant ...
can we just name it `DeepLink`

File: src/core/components/deeplinking-link.jsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react"
+import PropTypes from "prop-types"
+
+export const DeepLinkingLink = ({ isDeepLinkingEnabled, path, text }) => {
+    return (
+        <a className="nostyle"
+          onClick={isDeepLinkingEnabled ? (e) => e.preventDefault() : null}
```
Comment (heldersepu) [inline src/core/components/deeplinking-link.jsx]: Similar to my comment on the name...
Can we just simplify all these `isDeepLinkingEnabled` to just `enabled` we are already under the deep umbrella

**Comment by thompsongl** [comment]
Updated name to `DeepLink`

**Comment by thompsongl** [comment]
Anything else you'd like to see done here, @shockey?

**Comment by heldersepu** [comment]
What delay?

**Comment by shockey** [comment]
@heldersepu this PR was in my Demember 8 milestone 😉

**Comment by heldersepu** [comment]
@shockey It's December! no one really works on December...

**Comment by shockey** [comment]
@heldersepu 

# Can't have any new bugs if you don't write any code

roll-safe-meme-kayode-ewumi_bbc-three

**Pull request merged**